File: app/src/main/java/org/metabrainz/mobile/api/data/Track.java
package org.metabrainz.mobile.api.data;

import org.metabrainz.mobile.api.util.StringFormat;

public class Track {

    private String title;
    private String recordingMbid;
    private int position;
    private int duration;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getRecordingMbid() {
        return recordingMbid;
    }

    public void setRecordingMbid(String recordingMbid) {
        this.recordingMbid = recordingMbid;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public String getFormattedDuration() {
        return StringFormat.formatDuration(duration);
    }

}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/Track.kt
package org.metabrainz.mobile.data.sources.api.entities;

import com.google.gson.annotations.SerializedName;

import org.metabrainz.mobile.data.sources.api.entities.mbentity.Recording;

public class Track {
    @SerializedName("id")
    private String mbid;
    private String title;
    private int position;
    private Recording recording;
    private long length;

    public long getLength() {
        return length;
    }

    public void setLength(long length) {
        this.length = length;
    }

    public String getMbid() {
        return mbid;
    }

    public void setMbid(String mbid) {
        this.mbid = mbid;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public Recording getRecording() {
        return recording;
    }

    public void setRecording(Recording recording) {
        this.recording = recording;
    }

    public String getDuration() {
        StringBuilder builder = new StringBuilder();
        long seconds = length / 1000;
        long minutes = seconds / 60;
        seconds = seconds % 60;
        builder.append(minutes).append(':');
        if (seconds < 10) builder.append('0');
        builder.append(seconds);
        return builder.toString();
    }
}
